drivers/bluetooth/hci/cyw43xxx.c

```diff
@@ -0,0 +1,300 @@
+		uart_cfg.baudrate = baudrate;
+		err = uart_configure(bt_uart_dev, &uart_cfg);
+		if (err) {
+			return err;
+		}
+
+		/* Revert Interrupt options */
+		uart_irq_rx_enable(bt_uart_dev);
+	}
+	return 0;
+}
+
+static int bt_update_controller_baudrate(const struct device *bt_uart_dev, uint32_t baudrate)
+{
+	/*
+	 *  NOTE from datasheet for update baudrate:
+	 *  - To speed up application downloading, the MCU host commands the CYWxxx device
+	 *    to communicate at a new, higher rate by issuing the following Vendor Specific
+	 *    UPDATE_BAUDRATE command:
+	 *      01 18 FC 06 00 00 xx xx xx xx
+	 *    In the above command, the xx xx xx xx bytes specify the 32-bit little-endian
+	 *    value of the new rate in bits per second. For example,
+	 *    115200 is represented as 00 C2 01 00.
+	 *  The following response to the UPDATE_BAUDRATE command is expected within 100 ms:
+	 *  04 0E 04 01 18 FC 00
+	 *  - The host switches to the new baud rate after receiving the response at the old
+	 *  baud rate.
+	 */
+	struct net_buf *buf;
+	int err;
+	uint8_t hci_data[HCI_VSC_UPDATE_BAUD_RATE_LENGTH];
+
+	/* Baudrate is loaded LittleEndian */
+	hci_data[0] = 0;
+	hci_data[1] = 0;
+	hci_data[2] = (uint8_t)(baudrate & 0xFFUL);
+	hci_data[3] = (uint8_t)((baudrate >> 8) & 0xFFUL);
+	hci_data[4] = (uint8_t)((baudrate >> 16) & 0xFFUL);
+	hci_data[5] = (uint8_t)((baudrate >> 24) & 0xFFUL);
+
+	/* Allocate buffer for update uart baudrate command.
+	 * It will be BT_HCI_OP_RESET with extra parameters.
+	 */
+	buf = bt_hci_cmd_create(BT_HCI_VND_OP_UPDATE_BAUDRATE,
+				HCI_VSC_UPDATE_BAUD_RATE_LENGTH);
+	if (buf == NULL) {
+		return -1;
+	}
+
+	/* Add data part of packet */
+	void *data_buf = net_buf_add(buf, HCI_VSC_UPDATE_BAUD_RATE_LENGTH);
+
+	/* Copy data */
+	(void)memcpy(data_buf, (void *)hci_data, HCI_VSC_UPDATE_BAUD_RATE_LENGTH);
+
+	/* Send update uart baudrate command. */
+	err = bt_hci_cmd_send_sync(BT_HCI_VND_OP_UPDATE_BAUDRATE, buf, NULL);
+	if (err) {
+		return err;
+	}
+
+	/* Re-configure Uart baudrate */
+	err = bt_hci_uart_set_baudrate(bt_uart_dev, baudrate);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+static int bt_firmware_download(const uint8_t *firmware_image, uint32_t size)
+{
+	uint8_t *data = (uint8_t *) firmware_image;
+	volatile uint32_t remaining_length = size;
+	struct net_buf *buf;
+	int err;
+
+	LOG_DBG("Executing Fw downloading for CYW43xx device");
+
+	/* Send hci_download_minidriver command */
+	err = bt_hci_cmd_send_sync(BT_HCI_VND_OP_DOWNLOAD_MINIDRIVER, NULL, NULL);
+	if (err) {
+		return err;
+	}
+
+	/* The firmware image (.hcd format) contains a collection of hci_write_ram
+	 * command + a block of the image, followed by a hci_write_ram image at the end.
+	 * Parse and send each individual command and wait for the response. This is to
+	 * ensure the integrity of the firmware image sent to the bluetooth chip.
+	 */
+	while (remaining_length) {
+		size_t data_length = data[2]; /* data length from firmware image block */
```

Comment:
I'd suggest adding a debug log line in the cycle, this hangs after the first command if the firmware is incorrect, may come in handy. Up to you.